modules/apps/foundation/users-admin/users-admin-api/src/main/java/com/liferay/users/admin/UserInitialsGeneratorImpl.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.users.admin;
+
+import com.liferay.portal.kernel.language.LanguageConstants;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.util.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.users.admin.kernel.util.UserInitialsGenerator;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Pei-Jung Lan
+ */
+@Component(immediate = true, service = UserInitialsGenerator.class)
+public class UserInitialsGeneratorImpl implements UserInitialsGenerator {
+
+	@Override
+	public String getInitials(User user) {
```

Comment:
This looks as a very good method to write unit tests for. 

My suggestion, to make writing the test easier, is that you also add a method that receives the individual parameters (locale, firstName, middleName, lastName) instead of the actual User object. 

What do you think?